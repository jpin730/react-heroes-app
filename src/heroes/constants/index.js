export * from "./baseUrl";
export * from "./publishers";
