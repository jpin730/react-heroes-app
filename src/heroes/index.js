export * from "./components";
export * from "./constants";
export * from "./data";
export * from "./helpers";
export * from "./pages";
export * from "./routes";
