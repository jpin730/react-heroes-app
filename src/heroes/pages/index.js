export * from "./DcComicsPage";
export * from "./HeroPage";
export * from "./HomePage";
export * from "./MarvelPage";
export * from "./SearchPage";
