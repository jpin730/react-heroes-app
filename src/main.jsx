import React from "react";
import ReactDOM from "react-dom/client";
import { BrowserRouter } from "react-router-dom";

import { HeroesApp } from "./HeroesApp";
import "./styles.css";

ReactDOM.createRoot(document.getElementById("root")).render(
  <BrowserRouter basename={import.meta.env.PROD ? "/react-heroes-app" : "/"}>
    <React.StrictMode>
      <HeroesApp />
    </React.StrictMode>
  </BrowserRouter>
);
