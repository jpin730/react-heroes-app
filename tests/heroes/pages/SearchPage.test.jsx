import { render, screen } from "@testing-library/react";

import { SearchPage } from "../../../src/heroes";
import { MemoryRouter } from "react-router-dom";

describe("SearchPage", () => {
  test("should render with default values", () => {
    const { container } = render(
      <MemoryRouter initialEntries={["/search"]}>
        <SearchPage />
      </MemoryRouter>
    );

    expect(container).toMatchSnapshot();
    expect(screen.getByText("Search a hero")).toBeTruthy();
  });

  test("should render results", () => {
    const { container } = render(
      <MemoryRouter initialEntries={["/search?query=batman"]}>
        <SearchPage />
      </MemoryRouter>
    );

    expect(screen.getByRole("img").src).toContain("dc-batman.jpg");
    expect(screen.getByRole("heading", { level: 5 }).innerHTML).toContain(
      "Batman"
    );
  });

  test("should render alert on no results", () => {
    const { container } = render(
      <MemoryRouter initialEntries={["/search?query=invalid-hero-name"]}>
        <SearchPage />
      </MemoryRouter>
    );

    expect(screen.getByText("No hero with name")).toBeTruthy();
    expect(screen.getByText('"invalid-hero-name"')).toBeTruthy();
  });
});
