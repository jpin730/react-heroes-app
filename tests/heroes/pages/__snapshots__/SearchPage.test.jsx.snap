// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchPage should render with default values 1`] = `
<div>
  <h1>
    Search
  </h1>
  <hr />
  <div
    class="row g-4"
  >
    <div
      class="col-xl-4"
    >
      <form
        class="d-flex gap-3"
      >
        <input
          autocomplete="off"
          class="form-control"
          name="searchText"
          placeholder="Enter a hero name"
          type="text"
          value=""
        />
        <button
          class="btn btn-primary"
          disabled=""
          type="submit"
        >
          Search
        </button>
      </form>
    </div>
    <div
      class="col-xl-8"
    >
      <div
        class="alert alert-info animate__animated animate__fadeIn"
      >
        Search a hero
      </div>
      <div
        class="row row-cols-1 row-cols-lg-2 g-4"
      />
    </div>
  </div>
</div>
`;
